### DISCUSSION TITLE
Add PublishReadyToRun to DotNetRestoreSettings

### DISCUSSION THREAD

**Discussion by Trisped** [cake-build/cake#4108] [Ideas]
According to https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/6.0/publish-readytorun-requires-restore-change#recommended-action :
> In .NET 6, dotnet restore followed by dotnet publish -p:PublishReadyToRun=true --no-restore will fail with the NETSDK1095 error. This is because the crossgen binary is now shipped as a separate NuGet package, and so needs to be part of the restore operation for publishing to succeed.

According to https://cakebuild.net/api/Cake.Common.Tools.DotNet.Restore/DotNetRestoreSettings/ , there is not a PublishReadyToRun property like there is in DotNetPublishSettings.

There is a work around (set NoRestore to false), but it requires restoring twice, if you have already restored (for example, so you could test before publishing).

**Comment by augustoproiete** [comment]
Thanks @Trisped. I created https://github.com/cake-build/cake/issues/4109 to track this. Feels quite weird to have a parameter `PublishReadyToRun` on a `Restore` command, but hey... I know you're just the messenger 😃 

In the meantime, a workaround is to use the `ArgumentCustomization` that every `DotNet****` command settings has... E.g.:

```
DotNetRestore("./Project.csproj", new DotNetRestoreSettings
{
    ArgumentCustomization = args =>
        args.AppendQuoted($"-p:PublishReadyToRun=true")
});
```
